pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -129,11 +138,30 @@ public FieldSpec(String name, DataType dataType, boolean isSingleValueField, @Nu
 
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
       @Nullable Object defaultNullValue) {
+    this(name, dataType, isSingleValueField, maxLength, defaultNullValue, null);
+  }
+
+  public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
+      @Nullable Object defaultNullValue, @Nullable MaxLengthExceedStrategy maxLengthExceedStrategy) {
     _name = name;
     _dataType = dataType;
     _isSingleValueField = isSingleValueField;
     _maxLength = maxLength;
     setDefaultNullValue(defaultNullValue);
+    switch (_dataType) {
```

Comment:
We don't need to set it here. We can simply assign `_maxLengthExceedStrategy = maxLengthExceedStrategy`. Basically `FieldSpec.getMaxLengthExceedStrategy()` can return `null`, and `SanitizationTransformer` knows what default value to use when it is not explicitly configured